### PR TITLE
Issue #4080: Split and Organize Checkstyle inputs by Test for SingleSpaceSeparator

### PR DESCRIPTION
**Pull request by wate123**
Issue: #4080

### CODE REVIEW

**Comment by Vladlis** [comment]
We have https://github.com/checkstyle/checkstyle/wiki/PR-rules , please follow

**Comment by wate123** [comment]
how can I pass the Travis CI build?

**Comment by rnveach** [comment]
@wate123 You can click on `Details` for travis and view why it is failing.
https://travis-ci.org/checkstyle/checkstyle/jobs/214427252#L214
> You are providing a PR for an Issue that is not approved yet, please ask admins to approve your Issue first

Issue was never approved. It is fixed now.
Restarting travis.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/singlespaceseparator/InputSingleSpaceComments.java [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package com.puppycrawl.tools.checkstyle.checks.whitespace;
+package com.puppycrawl.tools.checkstyle.checks.whitespace.singlespaceseparator;
 
 public class InputSingleSpaceComments {
```
Comment (rnveach) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/singlespaceseparator/InputSingleSpaceComments.java]: Inputs should be named after the test. Test is `SingleSpaceSeparatorCheckTest`, so input names should start with `InputSingleSpaceSeparator`.
Please do this for all inputs.

**Comment by romani** [comment]
wercker problem should be resolved by rebasing on our latest master code.

Travis problems are your problems, please address. Run "mvn verify" on local to reproduce the same problem on you local.

**Comment by wate123** [comment]
Please help me out. every time I change the _InputSingleSpaceErrors_ to `InputSingleSpaceSeparatorErrors` I can't pass the Test. This bothers me for 2 days.

Failed tests: 
  SingleSpaceSeparatorCheckTest.testSpaceErrors:88->BaseCheckTestSupport.verify:191->BaseCheckTestSupport.verify:227->BaseCheckTestSupport.verify:258 error message 28 expected:<...aratorErrors.java:6:[42]: Use a single space...> but was:<...aratorErrors.java:6:[51]: Use a single space...>

Tests run: 2005, Failures: 1, Errors: 0, Skipped: 0

**Comment by rnveach** [comment]
We can only see what you have pushed to the server, and all tests are passing right now.

> <...aratorErrors.java:6:[42]: Use a single space...> but was:<...aratorErrors.java:6:[51]: Use a single space...>

It looks like issue is with `InputSingleSpaceErrors` file as violations are around class name. To change class name involves change violation positions.
@romani You ok with this or you want to keep class and violations the same?

**Comment by romani** [comment]
lets keep file name as is for now. 
moving file to separate folder is already big deal that ease management of them.

File: README.md [hunk 2]
```diff
@@ -74,8 +74,8 @@ directory.
 [appveyor]:https://ci.appveyor.com/project/checkstyle/checkstyle/history
 [appveyor img]:https://ci.appveyor.com/api/projects/status/rw6bw3dl9kph6ucc?svg=true
 
-[sonar]:https://sonarqube.com/overview?id=com.puppycrawl.tools%3Acheckstyle
-[sonar img]:https://img.shields.io/sonar/http/sonarqube.com/com.puppycrawl.tools:checkstyle/tech_debt.svg?label=Sonarqube%20tech%20debt
+[sonar]:https://sonarqube.com/dashboard?id=com.puppycrawl.tools%3Acheckstyle
+[sonar img]:https://sonarqube.com/api/badges/measure?key=com.puppycrawl.tools:checkstyle&metric=sqale_debt_ratio
```
Comment (rnveach) [inline README.md]: Changes to this file are not expected. Please revert them.

**Comment by rnveach** [comment]
@wate123 commits must stay at 1. We will not merge multiple commits for this.

File: README.md [hunk 3]
```diff
@@ -11,8 +11,8 @@
 [![][versioneye img]][versioneye]
 [![][sonar img]][sonar]
 
-Members chat: [![][gitter_mem img]][gitter_mem]
-Contributors chat: [![][gitter_con img]][gitter_con]
+Members chat: [![][gitter img]][gitter]
```
Comment (rnveach) [inline README.md]: Still should be no changes in this file

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/singlespaceseparator/InputSingleSpaceErrors.java [hunk 4]
```diff
@@ -1,4 +1,4 @@
-package  com.puppycrawl.   tools.checkstyle.checks.whitespace;
+package  com.puppycrawl.   tools.checkstyle.checks.whitespace.singlespaceseparator;
```
Comment (rnveach) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/singlespaceseparator/InputSingleSpaceErrors.java]: file was not renamed. It should be `InputSingleSpaceSeparatorErrors`.
Class name can't be changed as stated before.

**Comment by rnveach** [comment]
@romani I should be fine to merge after this last fix.

**Comment by rnveach** [comment]
@wate123 CI is failing, please run `mvn verify` before pushing changes to make sure everything is good.
There are alot of changes in `naming` folder. We expect changes only for `SingleSpaceSeparatorCheckTest` and it's inputs. All others need to be reverted.

File: README.md (hunk 2 above)
Comment (wate123) [inline README.md]: fixed it!

**Comment by wate123** [comment]
`mvn verify` still gave the fail test about the using single space in SIngleSpaceSeparatorErrors.java

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/singlespaceseparator/InputSingleSpaceSeparatorErrors.java [hunk 5]
```diff
@@ -1,9 +1,9 @@
-package  com.puppycrawl.   tools.checkstyle.checks.whitespace;
+package  com.puppycrawl.   tools.checkstyle.checks.whitespace.singlespaceseparator;
 
 import java.util.List;
 import  java.util.Vector;
 
-public class      InputSingleSpaceErrors  {
+public class      InputSingleSpaceSeparatorErrors  {
```
Comment (rnveach) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/singlespaceseparator/InputSingleSpaceSeparatorErrors.java]: @wate123 Maybe there is some confusion about what I am asking.
File should be renamed to `InputSingleSpaceSeparatorErrors.java`.
Class name must remain as `InputSingleSpaceErrors`.

My previous review was only pointing out that the file was not renamed.

Comment (wate123) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/singlespaceseparator/InputSingleSpaceSeparatorErrors.java]: Ummm. After I did this, It gives me a `COMPILATION ERROR` and told me to change the file name to InputSingleSpaceErrors.java

Comment (rnveach) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/singlespaceseparator/InputSingleSpaceSeparatorErrors.java]: > [ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.6.1:testCompile (default-testCompile) on project checkstyle: Compilation failure
[ERROR] /D:/checkstyleWorkspaceEclipse/checkstyle/src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/singlespaceseparator/InputSingleSpaceSeparatorErrors.java:[6,8] class InputSingleSpaceErrors is public, should be declared in a file named InputSingleSpaceErrors.java

You are correct. I didn't think the compiler would require this and make it an error.
I am sorry for the trouble.

@wate123 Please keep this file and it's class as its original name.
@romani Just so we can get this done, we can change class, file, and violation in new PR when the main issue is closer to being done.

Comment (wate123) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/singlespaceseparator/InputSingleSpaceSeparatorErrors.java]: What should I do next?

**Comment by rnveach** [comment]
@wate123 Like the PR rules say https://github.com/checkstyle/checkstyle/wiki/PR-rules
keep all commits to 1, and rebase off the latest master.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java [hunk 6]
```diff
@@ -207,12 +207,6 @@ else if (shouldBeOnSameLine(option, details)) {
         else if (shouldBeAloneOnLine(option, details)) {
             violation = MSG_KEY_LINE_ALONE;
         }
-        else if (shouldStartLine) {
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java]: Revert all changes that are not in `SingleSpaceSeparatorCheckTest` or it's inputs.

**Comment by rnveach** [comment]
@wate123 ping
Since we are out of proposal deadline, do you think you can finish this?

**Comment by wate123** [comment]
I accidentally close the pull request. how can I re open?

**Comment by rnveach** [comment]
@wate123 restore the branch, and you should be able to re-open it.

**Comment by wate123** [comment]
I can't restore now. I think I delete my branch and recreated one.

**Comment by rnveach** [comment]
@wate123 I see you did recreate the branch under the same name. I can't re-open it too.
You will probably have to create a new PR.
Please don't delete the branch until after the fix is merged.

**Comment by wate123** [comment]
@rnveach sorry about that. I will not doing this again!

**Pull request closed without merging**